src/Controls/src/Xaml/Hosting/AppHostBuilderExtensions.cs

```diff
@@ -1,260 +1,49 @@
+	/// <returns>The configured <see cref="MauiAppBuilder"/>.</returns>
+	public static MauiAppBuilder UseMauiApp<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TApp>(this MauiAppBuilder builder, Func<IServiceProvider, TApp> implementationFactory)
+		where TApp : class, IApplication
+	{
+		builder.UseMauiPrimaryApp<TApp>(implementationFactory);
+		builder.SetupXamlDefaults();
+		return builder;
+	}
 
-#if WINDOWS
-			handlersCollection.AddHandler<ShellItem, ShellItemHandler>();
-			handlersCollection.AddHandler<ShellSection, ShellSectionHandler>();
-			handlersCollection.AddHandler<ShellContent, ShellContentHandler>();
-			handlersCollection.AddHandler<Shell, ShellHandler>();
-#endif
-			return handlersCollection;
-		}
+	public static IMauiHandlersCollection AddMauiControlsHandlers(this IMauiHandlersCollection handlersCollection) =>
+		handlersCollection.AddControlsHandlers();
 
-		static MauiAppBuilder SetupDefaults(this MauiAppBuilder builder)
-		{
+	static MauiAppBuilder SetupXamlDefaults(this MauiAppBuilder builder)
+	{
 #if WINDOWS || ANDROID || IOS || MACCATALYST || TIZEN
-			// initialize compatibility DependencyService
-			DependencyService.SetToInitialized();
-			DependencyService.Register<Xaml.ResourcesLoader>();
-			DependencyService.Register<Xaml.ValueConverterProvider>();
-			DependencyService.Register<PlatformSizeService>();
-
-#pragma warning disable CS0612, CA1416 // Type or member is obsolete, 'ResourcesProvider' is unsupported on: 'iOS' 14.0 and later
-			DependencyService.Register<ResourcesProvider>();
-			DependencyService.Register<FontNamedSizeService>();
-#pragma warning restore CS0612, CA1416 // Type or member is obsolete
-#endif
-			builder.Services.AddScoped(_ => new HideSoftInputOnTappedChangedManager());
-			builder.ConfigureImageSourceHandlers();
-			builder
-				.ConfigureMauiHandlers(handlers =>
-				{
-					handlers.AddMauiControlsHandlers();
-				});
-
-#if WINDOWS
-			builder.Services.TryAddEnumerable(ServiceDescriptor.Transient<IMauiInitializeService, MauiControlsInitializer>());
-#endif
-			builder.RemapForControls();
-
-			return builder;
-		}
-
-		class MauiControlsInitializer : IMauiInitializeService
-		{
-			public void Initialize(IServiceProvider services)
-			{
-#if WINDOWS
-				var dispatcher = services.GetRequiredApplicationDispatcher();
-
-				dispatcher
-					.DispatchIfRequired(() =>
-					{
-						var dictionaries = UI.Xaml.Application.Current?.Resources?.MergedDictionaries;
-						if (dictionaries != null)
-						{
-							// Microsoft.Maui.Controls
-							UI.Xaml.Application.Current?.Resources?.AddLibraryResources("MicrosoftMauiControlsIncluded", "ms-appx:///Microsoft.Maui.Controls/Platform/Windows/Styles/Resources.xbf");
-						}
-					});
+		DependencyService.Register<Xaml.ResourcesLoader>();
+		DependencyService.Register<Xaml.ValueConverterProvider>();
```

Comment:
This file is the one I moved into Controls.Core - except for these 2 lines. The `UseMauiApp` API is still here, but now just calls into Controls.Core and then calls this one (`SetupXamlDefaults`).